### PR TITLE
[Bugfix] Fix duplicated KV cache allocation in qwen3-Next

### PR DESCRIPTION
**Pull request by QilaiZhang**
### What this PR does / why we need it?
Fix the bug described in https://github.com/vllm-project/vllm-ascend/issues/3368: Duplicated Allocation of Shared Hybrid Cache in Qwen3-Next

### Does this PR introduce _any_ user-facing change?
No
### How was this patch tested?
The model qwen3-next has been tested.

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.0

### CODE REVIEW

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -2850,49 +2850,33 @@ def initialize_kv_cache_tensors(
...
-                                                 device=self.device)
-                        else:
-                            cache_size_aligned = kv_cache_tensor.size + alignment
-                            tensor = torch.zeros(cache_size_aligned,
-                                                 dtype=torch.int8,
-                                                 device=self.device)
-                            tensor = self._align_memory(
-                                tensor, alignment)[:kv_cache_tensor.size]
-                        kv_cache_raw_tensors[layer_name_inner] = tensor
+                    kv_cache_raw_tensors[layer_name] = tensor
```
Comment (MengqingCao) [inline vllm_ascend/worker/model_runner_v1.py:2877]: Sharing kv cache memory between `linear_attn` and `self_attn` will cause accuracy issue in Qwen3-Next. Thus we should allocate memory for `linear_attn` and `self_attn` seperately.
Actually I also have a PR on fixing this issue and refactor the kvcache initialization logic, but it is not ready for review now. Feel free to change the bug fix logic in this pr according to #3106

**Comment by MengqingCao** [comment]
@QilaiZhang plz take a look at https://github.com/vllm-project/vllm-ascend/pull/3760, which shared kvcache in the same attention type, before fixing accuracy issue on sharing buffer between `linear_attn` and `self_attn`, I prefer to waste some memory to guarantee the accuracy

**Comment by QilaiZhang** [comment]
> @QilaiZhang plz take a look at #3760, which shared kvcache in the same attention type, before fixing accuracy issue on sharing buffer between `linear_attn` and `self_attn`, I prefer to waste some memory to guarantee the accuracy

@MengqingCao Thanks for flagging this. Acknowledged - we'll preserve the current memory usage to guarantee accuracy. We'll look into #3760 and revisit buffer sharing once the accuracy issues are fixed.
